### PR TITLE
Add date device class

### PR DESCRIPTION
**Pull request by scop**
## Proposed change

See https://github.com/home-assistant/architecture/discussions/610

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/55887
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/sensor.markdown [hunk 1]
```diff
@@ -22,6 +22,7 @@ The type of data a sensor returns impacts how it is displayed in the frontend. T
 - **carbon_dioxide**: Carbon Dioxide in CO2 (Smoke)
 - **carbon_monoxide**: Carbon Monoxide in CO (Gas CNG/LPG)
 - **current**: Current in A.
+- **date**: Date object or date string (ISO 8601).
```
Comment (MartinHjelmare) [inline source/_integrations/sensor.markdown]: The state returned from the state machine will always be an isoformatted date string. The incoming data that we convert in the sensor platform could be a date object.

I'm not sure what case this list is describing.

Comment (frenck) [reply source/_integrations/sensor.markdown]: The state, so it is a date string.

```suggestion
- **date**: Date string (ISO 8601).
```

**Review by frenck — APPROVED** [review]

**Comment by scop** [comment]
Thanks! JFYI, the reason for the original wording was that it was basically pasted from `timestamp` which still reads
```
**timestamp**: Datetime object or timestamp string (ISO 8601).
```
I admit not having done homework at all wrt this but just assumed that it'd behave similarly. If it doesn't, the doc for `timestamp` should probably be adjusted like it was done for `date` here by @frenck

**Comment by frenck** [comment]
Yeah, that probably makes sense. Most of these texts originate from the developer documentation (which is providing information versus the end-user documentation which consumes the result).

**Pull request merged**